src/types/datetime.rs

```diff
@@ -26,8 +25,64 @@ use crate::ffi::{
 use crate::types::PyTuple;
 use crate::{AsPyPointer, PyAny, PyObject, Python, ToPyObject};
 use std::os::raw::c_int;
-#[cfg(not(PyPy))]
-use std::ptr;
+
+fn ensure_datetime_api(_py: Python) -> &'static PyDateTime_CAPI {
+    unsafe {
+        if pyo3_ffi::PyDateTimeAPI().is_null() {
+            PyDateTime_IMPORT()
+        }
+
+        &*pyo3_ffi::PyDateTimeAPI()
+    }
+}
+
+/// Type Check macros
```

Comment:
This doesn't seem to be ta docstring for the macro?

Reply:
I added this doc comment to the macro because it seemed the best place to put it (it describes the function the macro generates) but I can make it a comment instead if you think that would be clearer

Reply:
Well, in the rendered docs it will show up as doc for the macro called ffi_fun_with_autoinit which seems to have nothing to do with type checking.

Reply:
Okay I will put it as a comment then so its lets confusing in the docs